templates/docs/document.html

```diff
@@ -1,22 +1,89 @@
-    {{ document.body_html | safe }}
-  </div>
-</div>
+  <div class="row">
+    <aside class="col-3">
+      <nav class="p-side-navigation--raw-html" id="drawer">
+        <button class="p-side-navigation__toggle js-drawer-toggle" aria-controls="drawer">
+          Toggle side navigation
+        </button>
+        <div class="p-side-navigation__overlay js-drawer-toggle" aria-controls="drawer"></div>
 
-<div class="p-strip is-shallow">
-  <div class="p-content__row">
-    <div class="p-notification--information">
-      <p class="p-notification__response">
-          Last updated {{ document.updated }}. <a href="{{ forum_url }}{{ document.topic_path }}">Help improve this document in the forum</a>.
-      </p>
-    </div>
+        <div class="p-side-navigation__drawer">
+          <div class="p-side-navigation__drawer-header">
+            <button class="p-side-navigation__toggle--in-drawer js-drawer-toggle" aria-controls="drawer">
+              Toggle side navigation
+            </button>
+          </div>
+          {{ navigation | safe }}
+        </div>
+      </nav>
+    </aside>
+
+    <main class="col-9" id="main-content">
+      <h1>{{ document.title }}</h1>
+      {{ document.body_html | safe }}
+      <hr />
+      <p><i>Last updated {{ document.updated }}.</i></p>
+      <div class="p-notification--information">
+        <p class="p-notification__response">
+          <a href="{{ forum_url }}{{ document.topic_path }}">Help improve this document in the forum</a>.
+        </p>
+      </div>
+    </main>
   </div>
 </div>
+
+<script>
+  /**
+    Toggles the expanded/collapsed classed on side navigation element.
+    @param {HTMLElement} sideNavigation The side navigation element.
+    @param {Boolean} show Whether to show or hide the drawer.
+  */
+  function toggleDrawer(sideNavigation, show) {
+    if (sideNavigation) {
+      if (show) {
+        sideNavigation.classList.remove('is-collapsed');
+        sideNavigation.classList.add('is-expanded');
+      } else {
+        sideNavigation.classList.remove('is-expanded');
+        sideNavigation.classList.add('is-collapsed');
+      }
+    }
+  }
+
+  /**
+    Attaches event listeners for the side navigation toggles
+    @param {HTMLElement} sideNavigation The side navigation element.
+  */
+  function setupSideNavigation(sideNavigation) {
+    var toggles = [].slice.call(sideNavigation.querySelectorAll('.js-drawer-toggle'));
+
+    toggles.forEach(function (toggle) {
+      toggle.addEventListener('click', function (event) {
+        event.preventDefault();
+        var sideNav = document.getElementById(toggle.getAttribute('aria-controls'));
+
+        if (sideNav) {
+          toggleDrawer(sideNav, !sideNav.classList.contains('is-expanded'));
+        }
+      });
+    });
+  }
+
+  var sideNav = document.getElementById("drawer");
+
+  setupSideNavigation(document.getElementById("drawer"));
+
+  // Add active links to active nav item and scroll to section
```

Comment:
Was this copied over from somewhere?
I just noticed that this comment is not relevant for the function below it. Maybe I left it by mistake somewhere else (snapcraft, dqlite)?